src/test/java/webApp/webControllerTest.java

```diff
@@ -17,11 +19,42 @@
 
 @SpringBootTest
 @AutoConfigureMockMvc
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
 public class webControllerTest {
 
     @Autowired
     private MockMvc mockMvc;
 
+    @Test
+    public void testRetrieveSurveyNames() throws Exception {
+        String surveyString = "{ \"name\" : \"survey1\", \"questions\" : [{ \"type\": \"openEnded\", \"question\": \"q1\" }, { \"type\": \"openEnded\", \"question\": \"q2\" }, { \"type\": \"numberQuestion\", \"question\": \"q3\", \"min\": \"0\", \"max\": \"5\" }]}";
+
+        this.mockMvc.perform(get("/retrieveSurveyNames")).andExpect(status().isOk())
+               .andExpect(content().json("{ \"nameList\":[], \"idList\" : []}"));
+
+        MvcResult result = this.mockMvc.perform(post("/createSurvey").contentType("application/json")
+                .content(surveyString)).andExpect(status().isOk())
+                .andExpect(content().string(containsString("ok")))
+                .andExpect(content().string(containsString("done")))
+                .andReturn();
+
+        Integer id1 = JsonPath.read(result.getResponse().getContentAsString(), "$.id");
```

Comment:
Any reason we use `Integer` here instead of the wrapper class `int`?

Reply:
Integer is the wrapper class.
No reason in particular.